### ISSUE TITLE
`:smt.mbqi false` & `(push)` changes check-sat result from unsat to unknown

### ISSUE THREAD

**Issue by bobismijnnaam** [Z3Prover/z3#6511]
Hi all, our tool turns off mbqi because of some oddities in our axiomatization. Now I've run into the problem where a `(push)` changes the result of a check-sat. Specifically, with `(push)` present, the result is unknown. With `(push)` commented out, the result is unsat.

```smt2
(set-option :smt.mbqi false)
(declare-sort P 0)
(declare-const someP P)

(push) ;; When commented out: unsat, otherwise: unknown

(assert (not (exists ((otherP P))
    (= otherP someP)
    )))
(check-sat)
```

Is there a technicality that causes this, or is it a bug? I tested this with 4.11.2.

EDIT: There is a duplicate issue: https://github.com/Z3Prover/z3/issues/341 I am not sure if was fully resolved back then, even though it mentions several short term fixes. I'll leave it to the maintainers to cut the knot.

**Comment by nunoplopes** [comment]
It's a "feature". Push changes the algorithms that are used for pre-processing. This is a known issue; may get fixed in the mid-term.
TL;DR: Avoid push if you can.
